File: src/lib/api/ask.ts
import { apiFetch } from "./client";

interface AskResponse {
  answer: string;
  deeplinks: { label: string; url: string }[];
  context_data: Record<string, unknown>;
}

export async function askQuestion(question: string): Promise<AskResponse> {
  return apiFetch<AskResponse>("/ask/", {
    method: "POST",
    body: JSON.stringify({ question }),
  });
}

File: backend/app/models/ask.py
from pydantic import BaseModel
from typing import Any


class AskRequest(BaseModel):
    question: str


class DeepLinkItem(BaseModel):
    label: str
    url: str


class AskResponse(BaseModel):
    answer: str
    deeplinks: list[DeepLinkItem] = []
    context_data: dict[str, Any] = {}
